Archive - new api part (#194)
digipost/digipost-api-client-dotnet: docs/_v14_0/4_archive.md

_`sender` is your sender, the one you use to create the client instance, if you are not a broker._ For broker see below.

```csharp
var archive = new Archive.Archive(sender, new List<ArchiveDocument>
{
    new ArchiveDocument(Guid.NewGuid(), "invoice_123123.pdf", "pdf","application/psd", readFileFromDisk("invoice_123123.pdf")),
    new ArchiveDocument(Guid.NewGuid(), "attachment_123123.pdf", "pdf","application/psd", readFileFromDisk("attachment_123123.pdf"))
});

var savedArchive = await client.GetArchive().ArchiveDocuments(archive);
```

You can name the archive if you wish to model your data in separate archives

```csharp
var archive = new Archive.Archive(sender, new List<ArchiveDocument>(), "MyArchiveName");
```

## Get a list of archives
An organisation can have many archives, or just the default unnamed archive. That is up to
your design wishes. To get a list of the archives for a given Sender, you can do this:

```csharp
IEnumerable<Archive.Archive> fetchedArchives = await client.GetArchive().FetchArchives();
```

## Iterate documents in an archive
You _can_ get content of an archive with paged requests. Under is an example of how to iterate

> an archive. However, it's use is strongly discouraged because it leads to the idea that

Reviewer: ```suggestion
an archive. However, its use is strongly discouraged because it leads to the idea that
```

_Please use fetch document by UUID or referenceID instead to create functionality on top of the archive._
You should on your side know where and how to get a document from an archive. You do this by knowing where
you put a file you want to retrieve.

```csharp
var current = (await client.GetArchive().FetchArchives()).First();
var documents = new List<ArchiveDocument>();

while (current.HasMoreDocuments())
{
    var fetchArchiveDocuments = client.GetArchive().FetchArchiveDocuments(current.GetNextDocumentsUri()).Result;
    documents.AddRange(fetchArchiveDocuments.ArchiveDocuments);
    current = fetchArchiveDocuments;
}

// documents now have all ArchiveDocuments in the archive (not the actual bytes, just the meta data)
```

In c# 8 there is a language feature to produce a `IAsyncEnumerable` so that you can implement a stream of 
archive documents. But this has not been implemented in this client library since we are on dotnetstandard2.0.
For each iteration above you will get 100 documents.

## Archive Document attributes
You can add optional attributes to documents. An attribute is a key/val string-dictionary that describe documents. You can add

> up to 15 attributes pr. archive document. The attribute key and value is case sensitive.

Reviewer: ```suggestion
up to 15 attributes per archive document. The attribute's key and value are case sensitive.
```
